demo.scenario

```diff
@@ -0,0 +1,147 @@
+$ # We can write everything down in default.nix
+$ bat default.nix
+────────┬────────────────────────────────────────────────────────────────────
+        │ File: default.nix
+────────┼──────────────────────────────────────────────────────────────────── 
+    1   │ { pkgs ? import <nixpkgs> {} # here we import nixpkgs package set
+    2   │ }:
+    3   │ pkgs.mkShell {               # mkShell is a helping function
+    4   │   name="dev-environment";    # that requires a name
+    5   │   buildInputs = [            # list of packages
+    6   │     pkgs.nodejs
+    7   │   ];
+    8   │   shellHook = ''             # bash to run when you enter the shell
+    9   │     echo "Start developing..."
+   10   │   '';
+   11   │ }
+────────┴────────────────────────────────────────────────────────────────────
+$ # Pause the video to understand the default.nix
+$ # To enter dev-environment simply run:
+$ nix-shell
+Start developing...
+(nix-shell) $ node -e "console.log(1+1)"
+2
+(nix-shell) $ # Now go ahead commit default.nix to your repository
+(nix-shell) $ # and share development environment with your coworkers
+
+--
+
+$ # For the last example let us build a minimal docker image with Nix
+$ bat docker-redis.nix
+───────┬─────────────────────────────────────────────────────────────────────
+       │ File: docker-redis.nix
+───────┼─────────────────────────────────────────────────────────────────────
+   1   │ { pkgs ? import <nixpkgs> {}         # nixpkgs package set
+   2   │ }:
+   3   │ pkgs.dockerTools.buildLayeredImage { # helper to build docker image
+   4   │   name = "nix-redis";                # give docker image a name
+   5   │   tag = "latest";                    # provide a tag
+   6   │   contents = [ pkgs.redis ];         # packages in docker image
+   7   │ }
+───────┴─────────────────────────────────────────────────────────────────────
+$ # Pause the video and take the time to understand docker-redis.nix file
+$ # Now lets build the docker image and load it to docker
+$ nix-build docker-redis.nix -o ./result 
+... SKIPPING OUTPUT ...
+/nix/store/1crapx24sjgqm2j1wmq17k6f6a9wy66d-docker-image-nix-redis.tar.gz
+$ docker load -i ./result   
+Loaded image: nix-redis:latest
+$ docker images | grep redis
+debian-redis  latest    8366943c77e8     3 days ago       136MB
+alpine-redis  latest    aae644cd3417     3 days ago       6.99MB
+nix-redis     latest    30486183a209     50 years ago     45.4MB
+$ # Size of a Nix based docker image is somewhere between a Debian and Alpine
+$ # redis packaged in nixpkgs is not optimized to be minimal as is in Alpine
+$ # Let us fix this!
+$ bat redis-minimal.nix
+───────┬─────────────────────────────────────────────────────────────────────
+       │ File: redis-minimal.nix
+───────┼─────────────────────────────────────────────────────────────────────
+   1   │ { pkgs ? import <nixpkgs> {}
+   2   │ }:
+   3   │ pkgs.redis.overrideAttrs (old: {
```

Comment:
This one looks less scary to the user and still makes use of `musl`

```suggestion
   3   │ pkgsStatic.redis.overrideAttrs (old: {
```

Than you would only need to override the `postInstall` phase.

Reply:
Mhm. seems to rebuild a lot more though.

Reply:
yeah, I tried `pkgsStatic` but it didn't build. It would be nice if it would be working.